src/cache.jl

```diff
@@ -127,6 +127,39 @@ end
                 true)))
 end
 
+disk_cache() = parse(Bool, @load_preference("disk_cache", "false"))
+cache_key() = @load_preference("cache_key", "")
+
+"""
+    enable_cache!(state=true)
+
+Activate the GPUCompiler disk cache in the current environment.
+You will need to restart your Julia environment for it to take effect.
+
+!!! warning
+    The disk cache is not automatically invalidated. It is sharded upon
+    `cache_key` (see [`set_cache_key``](@ref)), the GPUCompiler version
+    and your Julia version.
+"""
+function enable_cache!(state=true)
+    @set_preferences!("disk_cache"=>state)
+end
+
+"""
+    set_cache_key(key)
+
+If you are deploying an application it is recommended that you use your
+application name and version as a cache key. To minimize the risk of
+encountering spurious cache hits.
+"""
+function set_cache_key(key)
+    @set_preferences!("cache_key"=>key)
+end
+
+key(ver::VersionNumber) = "$(ver.major)_$(ver.minor)_$(ver.patch)"
+cache_path() = @get_scratch!(cache_key() * "-kernels-" * key(VERSION))
```

Comment:
Maybe include "cache" in the directory name? Or make this a subdirectory of the existing `compile_cache` scratch directory? That way the cache would also get wiped on `reset_runtime`, which is done when recompiling CUDA.jl. Or is that unwanted?

Reply:
Might also be confusing to have `compiled` in the scratch dir, containing the runtime bitcode, and `cache` for compiled kernels :-) maybe `cache/{runtime,jobs}`?

I know we're bikeshedding here :-)

Reply:
> That way the cache would also get wiped on reset_runtime, which is done when recompiling CUDA.jl. Or is that unwanted?

I was trying to add a dependency on the version of GPUCompiler/CUDA. Cache invalidation is a big potential footgun here.